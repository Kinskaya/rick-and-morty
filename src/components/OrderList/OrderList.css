.order-container {
    display: flex;
    flex-wrap: wrap;
    margin-top: 32px;
}

.order {
    width: 370px;
    height: auto;
    background: radial-gradient(109.56% 109.56% at 0% -2.94%, rgba(255, 255, 255, 0.5) 0%, rgb(217, 208, 248) 100%);
    border-radius: 10px;
    padding-left: 13px;
    margin: 0 0 30px 30px;
}

.order h3 {
    text-align: center;
    margin: 10px 0 0 0;
}

.order p {
    margin: 8px;
    padding: 0;
    font-weight: 500;
    opacity: 0.9;
    font-size: 0.97rem;
}
.order-image img {
    object-fit: contain;
    width: 95%;
    height: auto;
    display: block;
    margin-bottom: 10px;
    border-radius: 8px;
}

.order span {
    color: #414191;
    margin-left: 5px;
}
