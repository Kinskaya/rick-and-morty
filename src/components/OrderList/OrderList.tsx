import React from 'react';
import { useSelector } from 'react-redux';

import { TOrderForm } from '../../pages/OrderForm/OrderForm';
import { RootState } from '../../redux/store';
import { IState } from '../App/App';

import './OrderList.css';

export type TOrderList = {
  orders: TOrderForm[];
};

const OrderList: React.FunctionComponent<TOrderList> = () => {
  const { orders } = useSelector<RootState>((state) => state.orders) as IState;
  return (
    <div className="card-container">
      {orders.map((order, index) => {
        return (
          <div className="order" key={index}>
            <h3>Your order</h3>
            <p>
              Your Name:
              <span>{order.nameInput}</span>
            </p>
            <p>
              Delivery Date:
              <span>{order.dateInput}</span>
            </p>
            <p>
              Delivery Place:
              <span>{order.placeSelect}</span>
            </p>
            <p>
              Contact before delivery:
              <span>{order.reminder ? 'Yes' : 'No'}</span>
            </p>
            <p>
              I want to receive notifications about promo:
              <span>{order.isToggled ? 'Yes' : 'No'}</span>
            </p>
            <p>
              File:
              <span>{order.selectedFile?.name}</span>
            </p>
            <div className="order-image">
              <img
                src={URL.createObjectURL(order.selectedFile as File)}
                alt={order.selectedFile?.name}
              />
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default OrderList;
