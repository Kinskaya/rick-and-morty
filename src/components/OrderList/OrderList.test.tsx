import { render } from '@testing-library/react';
import OrderList from './OrderList';

describe('OrderList component', () => {
  it('render correctly', () => {
    window.URL.createObjectURL = jest.fn();
    const items = [
      {
        nameInput: 'Tom',
        placeSelect: 'Minsk',
        reminder: true,
        isToggled: true,
        dateInput: '2022-04-08',
        selectedFile: undefined,
      },
    ];
    const { asFragment } = render(<OrderList orders={items} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
