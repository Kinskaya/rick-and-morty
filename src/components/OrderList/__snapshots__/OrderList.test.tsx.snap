// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList component render correctly 1`] = `
<DocumentFragment>
  <div
    class="card-container"
  >
    <div
      class="order"
    >
      <h3>
        Your order
      </h3>
      <p>
        Your Name:
        <span>
          Tom
        </span>
      </p>
      <p>
        Delivery Date:
        <span>
          2022-04-08
        </span>
      </p>
      <p>
        Delivery Place:
        <span>
          Minsk
        </span>
      </p>
      <p>
        Contact before delivery:
        <span>
          Yes
        </span>
      </p>
      <p>
        I want to receive notifications about promo:
        <span>
          Yes
        </span>
      </p>
      <p>
        File:
        <span />
      </p>
      <div
        class="order-image"
      >
        <img />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
